ARCs/arc-draft_alternative_0014.md

```diff
@@ -0,0 +1,285 @@
+2. Bob generates a message `msg` for *PKa* and request Alice to sign it;
+3. Alice signs `msg` using her wallet, and afterwards sends back the tuple `<PKa, Sig(msg)>` to Bob;
+4. Bob verifies the digital signature `Sig(msg)` with the *PKa* and `msg`;
+5. If the signature is valid, then Bob generates a new `session-id` for Alice; otherwise returns NULL (authentication rejected);
+
+An ARC-0014 compliant message is called *Authentication Message*. A User **MUST** sign an *Authentication Message* with the *SKa* stored into a wallet.
+
+### Authentication Message
+
+An Authentication Message is a sequence of bytes representing a message. The Verifier asks Users to sign an *Authentication Message* with their secret keys. Such a message **MUST** include the following information:
+
+- `domain name` of the service;
+- `description` of the service;
+- `Algorand address` (PKa) to be authenticated;
+- `nonce` representing a unique/random value generated by the verifier.
+
+The JSON structure for such an object is:
+
+```typescript
+interface AuthMessage {
+ /** The domain name of the service */
+ service: string;
+ /** Optional, description of the service */
+ desc?: string;
+ /** Algorand account to authenticate with*/
+ authAcc: string;
+ /** Challenge generated server-side */
+ nonce: string;
+}
+```
+
+For example:
+
+```json
+{
+ "service": "www.servicedomain.com",
+ "desc": "Domain offers important services to users",
+ "authAcc": "KTGP47G64KCXWJS64W7SGJNKTHE37TYDCI64USXI3XOYE6ZSH4LCI7NIDA",
+ "nonce": "1234abcde!%",
+}
+```
+
+The `nonce` field **MUST** be unique for each authentication and **MUST NOT** be used more than once to avoid replay attacks.
+
+Users sign *Authentication Message*s using the secret keys stored into a wallet.
+
+***Problem***: Most of the Algorand wallets can only sign Algorand `Transaction` objects, and do not offer the possibility to sign arbitrary bytes. To overcome such a limitation, the ARC-0014 distinguishes two authentication objects, namely the **Simple Authentication Message**, and the **Transaction Authentication Message**. The former **SHOULD** be used by any signing mechanisms able to sign random bytes, whereas the latter can be used with almost any Algorand wallet.
+
+#### Simple Authentication Message
+
+The *Simple Authentication Message* is a sequence of bytes representing an ARC-0014 *Authentication Message*. It **SHOULD** be used in contexts that support signatures of random bytes. A *Simple Authentication Message* is an *Authentication Message* prepended with the prefix `ARC-0014-authentication` for domain separation. It **MUST** be represented as the hash SHA-512/256 of such prefix, together with a msgpack encoded *Authentication Message*. For example, given an *Authentication Message* object `aut_message`, its *Simple Authentication Message* representation would be something like: `SHA512_256(“ARC-0014-authentication”+msgpacked_auth_message)`.
```

Comment:
I think something shorter than ARC-0014-authentication would be sufficient, such as `arc14` see the domain separators here: https://github.com/algorand/go-algorand/blob/1504f5f39ec2893d2b0ac9b60d1b87f6e7d4e2fc/protocol/hash.go#L25